src/Bicep.Decompiler/BicepDecompiler.cs

```diff
@@ -58,9 +59,17 @@ public async Task<DecompileResult> Decompile(Uri entryJsonUri, Uri entryBicepUri
                 continue;
             }
 
-            if (!fileResolver.TryRead(jsonUri, out var jsonInput, out _))
+            var jsonInput = jsonTemplate;
+
+            if (jsonTemplate is null)
+            {
+                if (!fileResolver.TryRead(jsonUri, out jsonInput, out _))
+                {
+                    throw new InvalidOperationException($"Failed to read {jsonUri}");
+                }
+            } else
             {
-                throw new InvalidOperationException($"Failed to read {jsonUri}");
+                jsonInput = jsonTemplate;
```

Comment:
Happy to change this to a more elegant style instead of just repeating the declaration. This was the only way I found to make it work without line 75 complaining of a null reference

Reply:
Instead of adding a new optional parameter, we should consider creating an overload of the method (and potentially have this method reuse the overload).